aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -90,7 +90,7 @@ module aptos_framework::block {
         );
 
         let block_metadata_ref = borrow_global_mut<BlockMetadata>(@aptos_framework);
-        block_metadata_ref.height = block_metadata_ref.height + 1;
+        block_metadata_ref.height = event::counter(&block_metadata_ref.new_block_events);
```

Comment:
I'd keep the logic here but check where the event  is emitted. (You can make a mistake by updating the state but not emitting an event like this)

Reply:
I'm fine with making mistake (non-continuous height) than aborts in block prologue (network halt)

Reply:
Fine.
@gregnazario  shall we agree on seeing the event instead of the state (write set) as the source of truth?